bioamptool.py

```diff
@@ -221,104 +246,152 @@ def log_ten_minute_data(verbose=False):
+    except KeyboardInterrupt:
+        print("Process interrupted by user")
+    
+    finally:
+        if ser.is_open:
+            send_command(ser, 'STOP')
+            ser.reset_input_buffer()
+            ser.reset_output_buffer()
+            ser.close()
+
+        if lsl_outlet:     # Assuming LSL outlet cleanup if required
+            lsl_outlet = None
         
+        if csv_file:    # Ensure CSV file is closed
+            csv_file.close()
+            print(f"CSV recording saved as {csv_filename}")
+        
+        if gui_flag:    # Ensure GUI is properly closed
+            if timer:
+                timer.stop()
+            if app:
+                app.quit() 
+
+    print(f"Exiting.\nTotal missing samples: {missing_samples}")
+    sys.exit(0)
+
+def signal_handler(sig, frame):
+    print("Process interrupted by user")
+    global ser, csv_file, lsl_outlet, app  # Declare global variables
+    
+    ser = None
+    csv_file = None
+    lsl_outlet = None
+    app = None
+    
+    # Perform any necessary cleanup here
+    if ser and ser.is_open:
+        send_command(ser, 'STOP')
+        ser.reset_input_buffer()
+        ser.reset_output_buffer()
+        ser.close()
+    if lsl_outlet:
+        lsl_outlet = None  # Cleanup if needed
+    if csv_file:
+        csv_file.close()  # Close the CSV file
+        print(f"CSV recording saved as {csv_filename}")
+    if app:
+        app.quit()  # Close the GUI application
+    sys.exit(0)
+
 # Main entry point of the script
 def main():
     global verbose
     parser = argparse.ArgumentParser(description="Upside Down Labs - BioAmp Tool")  # Create argument parser
     parser.add_argument('-p', '--port', type=str, help="Specify the COM port")  # Port argument
-    parser.add_argument('-b', '--baudrate', type=int, default=57600, help="Set baud rate for the serial communication")  # Baud rate 
+    parser.add_argument('-b', '--baudrate', type=int, default=230400, help="Set baud rate for the serial communication")  # Baud rate 
     parser.add_argument('--csv', action='store_true', help="Create and write to a CSV file")  # CSV logging flag
     parser.add_argument('--lsl', action='store_true', help="Start LSL stream")  # LSL streaming flag
     parser.add_argument('--gui', action='store_true', help="Start GUI for real-time plotting")  # GUI flag
     parser.add_argument('-v', '--verbose', action='store_true', help="Enable verbose output with statistical data")  # Verbose flag
+    parser.add_argument('-t', '--time', type=int, help="Run the program for a specified number of seconds and then exit")   #set time
 
     args = parser.parse_args()  # Parse command-line arguments
     verbose = args.verbose  # Set verbose mode
 
     # Check if any logging or GUI options are selected, else show help
-    if not args.csv and not args.lsl and not args.gui:
+    if not args.csv and not args.lsl and not args.gui and args.time is None:
```

Comment:
why `args.time` is in here? we want CSV/LSL/GUI so that we have some way of getting the data not sure what time has to do with that. To me this code means that this will run even if we provide time argument which is not what we want, we need CSV/LSL/GUI argument to start.

Reply:
Resolved